File: src/main/java/io/github/chindeaone/collectiontracker/collections/BazaarCollectionsManager.java
package io.github.chindeaone.collectiontracker.collections;

import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import io.github.chindeaone.collectiontracker.collections.prices.BazaarPrices;
import io.github.chindeaone.collectiontracker.collections.prices.GemstonePrices;

import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;

public class BazaarCollectionsManager {

    public static boolean hasBazaarData = false;

    public static final Map<String, Integer> enchantedRecipe = new HashMap<>();
    public static final Map<String, Integer> superEnchantedRecipe = new HashMap<>();

    public static final Map<String, Map<String, Integer>> multiEnchantedRecipes = new HashMap<>();
    public static final Map<String, Map<String, Integer>> multiSuperEnchantedRecipes = new HashMap<>();

    public static void setPricesAndRecipes(String json, String type) {
        enchantedRecipe.clear();
        superEnchantedRecipe.clear();

        JsonObject jsonObject = JsonParser.parseString(json).getAsJsonObject();

        if (!jsonObject.has("prices") && !jsonObject.has("recipe")) {
            BazaarPrices.setPrices(jsonObject.toString(), type);
        } else {

            JsonObject prices = jsonObject.getAsJsonObject("prices");
            JsonObject recipe = jsonObject.getAsJsonObject("recipe");

            Iterator<Map.Entry<String, JsonElement>> iterator = recipe.entrySet().iterator();

            if (iterator.hasNext()) {
                Map.Entry<String, JsonElement> entry = iterator.next();
                enchantedRecipe.put(entry.getKey(), entry.getValue().getAsInt());
            }

            if (iterator.hasNext()) {
                Map.Entry<String, JsonElement> entry = iterator.next();
                superEnchantedRecipe.put(entry.getKey(), entry.getValue().getAsInt());
            }
            BazaarPrices.setPrices(prices.toString(), type);
        }
        hasBazaarData = true;
    }

    public static void setPricesAndRecipes(String collection, String json, String type) {
        JsonObject jsonObject = JsonParser.parseString(json).getAsJsonObject();

        if (jsonObject.has("prices") || jsonObject.has("recipe")) {
            JsonObject prices = jsonObject.getAsJsonObject("prices");
            JsonObject recipe = jsonObject.getAsJsonObject("recipe");

            Map<String, Integer> enchanted = new HashMap<>();
            Map<String, Integer> superEnchanted = new HashMap<>();

            Iterator<Map.Entry<String, JsonElement>> iterator = recipe.entrySet().iterator();
            if (iterator.hasNext()) {
                Map.Entry<String, JsonElement> entry = iterator.next();
                enchanted.put(entry.getKey(), entry.getValue().getAsInt());
            }
            if (iterator.hasNext()) {
                Map.Entry<String, JsonElement> entry = iterator.next();
                superEnchanted.put(entry.getKey(), entry.getValue().getAsInt());
            }

            multiEnchantedRecipes.put(collection, enchanted);
            multiSuperEnchantedRecipes.put(collection, superEnchanted);
            BazaarPrices.setPrices(collection, prices.toString(), type);
        } else {
            BazaarPrices.setPrices(collection, jsonObject.toString(), type);
        }
        hasBazaarData = true;
    }

    public static void resetBazaarData() {
        hasBazaarData = false;
        enchantedRecipe.clear();
        superEnchantedRecipe.clear();
        multiEnchantedRecipes.clear();
        multiSuperEnchantedRecipes.clear();
        BazaarPrices.resetPrices();
        GemstonePrices.resetPrices();
    }
}

File: src/main/java/io/github/chindeaone/collectiontracker/collections/BazaarCollectionsManager.kt
package io.github.chindeaone.collectiontracker.collections

import com.google.gson.JsonParser
import io.github.chindeaone.collectiontracker.collections.prices.BazaarPrices
import io.github.chindeaone.collectiontracker.collections.prices.BazaarPrices.setPrices
import io.github.chindeaone.collectiontracker.collections.prices.GemstonePrices

object BazaarCollectionsManager {
    var hasBazaarData: Boolean = false

    val enchantedRecipe: MutableMap<String, Int> = mutableMapOf()
    val superEnchantedRecipe: MutableMap<String, Int> = mutableMapOf()

    val multiEnchantedRecipes: MutableMap<String, MutableMap<String, Int>> = mutableMapOf()
    val multiSuperEnchantedRecipes: MutableMap<String, MutableMap<String, Int>> = mutableMapOf()

    fun setPricesAndRecipes(json: String, type: String) {
        enchantedRecipe.clear()
        superEnchantedRecipe.clear()

        val jsonObject = JsonParser.parseString(json).getAsJsonObject()

        if (!jsonObject.has("prices") && !jsonObject.has("recipe")) {
            setPrices(jsonObject.toString(), type)
        } else {
            val prices = jsonObject.getAsJsonObject("prices")
            val recipe = jsonObject.getAsJsonObject("recipe")

            val iterator = recipe.entrySet().iterator()

            if (iterator.hasNext()) {
                val entry = iterator.next()
                enchantedRecipe[entry.key] = entry.value.asInt
            }

            if (iterator.hasNext()) {
                val entry = iterator.next()
                superEnchantedRecipe[entry.key] = entry.value.asInt
            }
            setPrices(prices.toString(), type)
        }
        hasBazaarData = true
    }

    fun setPricesAndRecipes(collection: String, json: String, type: String) {
        val jsonObject = JsonParser.parseString(json).getAsJsonObject()

        if (jsonObject.has("prices") || jsonObject.has("recipe")) {
            val prices = jsonObject.getAsJsonObject("prices")
            val recipe = jsonObject.getAsJsonObject("recipe")

            val enchanted: MutableMap<String, Int> = mutableMapOf()
            val superEnchanted: MutableMap<String, Int> = mutableMapOf()

            val iterator = recipe.entrySet().iterator()
            if (iterator.hasNext()) {
                val entry = iterator.next()
                enchanted[entry.key] = entry.value.asInt
            }
            if (iterator.hasNext()) {
                val entry = iterator.next()
                superEnchanted[entry.key] = entry.value.asInt
            }

            multiEnchantedRecipes[collection] = enchanted
            multiSuperEnchantedRecipes[collection] = superEnchanted
            setPrices(collection, prices.toString(), type)
        } else {
            setPrices(collection, jsonObject.toString(), type)
        }
        hasBazaarData = true
    }

    fun resetBazaarData() {
        hasBazaarData = false
        enchantedRecipe.clear()
        superEnchantedRecipe.clear()
        multiEnchantedRecipes.clear()
        multiSuperEnchantedRecipes.clear()
        BazaarPrices.resetPrices()
        GemstonePrices.resetPrices()
    }
}